tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+			})
+	})
+})
+
+func updateMachineConfigurationCluster(nftables bool) {
+	By("should update machineconfiguration cluster")
+
+	var jsonBytes []byte
+
+	if nftables {
+		jsonBytes = []byte(`
+	{"spec":{"nodeDisruptionPolicy":
+	  {"files": [{"actions":
+	[{"restart": {"serviceName": "nftables.service"},"type": "Restart"}],
+	"path": "/etc/sysconfig/nftables.conf"}],
+	"units":
+	[{"actions":
+	[{"reload": {"serviceName":"nftables.service"},"type": "Reload"},
+	{"type": "DaemonReload"}],"name": "nftables.service"}]}}}`)
+	} else {
+		jsonBytes = []byte(`{"spec":{
+	"nodeDisruptionPolicy": {"files": [],"units": []}
+	}}`)
+	}
+
+	err := APIClient.Patch(context.TODO(), &ocpoperatorv1.MachineConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+	}, client.RawPatch(apimachinerytype.MergePatchType, jsonBytes))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = WaitForMcpStable(APIClient, 35*time.Minute, 1*time.Minute, "workercnf")
+	Expect(err).ToNot(HaveOccurred(), "Failed to wait for MCP to be stable")
+}
+
+func createExternalNad(name string) {
+	var externalNad *nad.Builder
+
+	By("Creating external BR-EX NetworkAttachmentDefinition")
+
+	macVlanPlugin, err := define.MasterNadPlugin(coreparams.OvnExternalBridge, "bridge", nad.IPAMStatic())
+	Expect(err).ToNot(HaveOccurred(), "Failed to define master nad plugin")
+	externalNad, err = nad.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithMasterPlugin(macVlanPlugin).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create external NetworkAttachmentDefinition")
+	Expect(externalNad.Exists()).To(BeTrue(), "Failed to detect external NetworkAttachmentDefinition")
+}
+
+func setupRemoteMultiHopTest(ipv4SecurityIPList, hubIPv4ExternalAddresses,
+	ipv4NodeAddrList []string, workerNodeList []*nodes.Builder, hubPodWorkerName []string, portNum int) *pod.Builder {
+	By("Creating External NAD for master FRR pod")
+	createExternalNad(tsparams.ExternalMacVlanNADName)
+
+	By("Creating External NAD for hub FRR pods")
+	createExternalNad(nadHub)
+
+	By("Creating static ip annotation for hub0")
+
+	hub0BRstaticIPAnnotation := createStaticIPAnnotations(tsparams.ExternalMacVlanNADName,
+		nadHub,
+		[]string{fmt.Sprintf("%s/24", ipv4SecurityIPList[0])},
+		[]string{fmt.Sprintf("%s/24", hubIPv4ExternalAddresses[0])})
+
+	By("Creating static ip annotation for hub1")
+
+	hub1BRstaticIPAnnotation := createStaticIPAnnotations(tsparams.ExternalMacVlanNADName, nadHub,
+		[]string{fmt.Sprintf("%s/24", ipv4SecurityIPList[1])},
+		[]string{fmt.Sprintf("%s/24", hubIPv4ExternalAddresses[1])})
+
+	By("Creating Frr Hub pod configMap")
+
+	hubConfigMap := createHubConfigMap("hub-node-config")
+
+	By("Creating FRR Hub pod on worker node 0")
+
+	_ = createFrrHubPod(hubPodWorkerName[0],
+		workerNodeList[0].Object.Name, hubConfigMap.Definition.Name, []string{}, hub0BRstaticIPAnnotation)
+
+	By("Creating FRR Hub pod on worker node 1")
```

Comment:
Why do you need frr pod for this test?

Reply:
I need it for the routing. I tried with many different configurations and was unable to get a packet to cross the nftables rule to be blocked.